src/components/details/PokemonDetails.tsx

```diff
@@ -1,15 +1,53 @@
 import { StatusBar } from 'expo-status-bar';
-import { SafeAreaView, StyleSheet, Text, View } from 'react-native';
+import { ActivityIndicator, SafeAreaView, StyleSheet, Text } from 'react-native';
+import React, { useState, useEffect } from 'react';
+import { View } from 'react-native';
 
 
-export function PokemonDetails({ route, navigation }: { route: any, navigation: any }) {
+
+export function PokemonDetails({ route }: { route: any, navigation: any }) {
     const { name, uri } = route.params;
 
+    const [isLoading, setIsLoading] = useState<boolean>(true);
```

Comment:
Look into using useQuery hook. It should do nearly all of this code block for you unless I am missing something https://tanstack.com/query/v4/docs/react/reference/useQuery